hngprojects/hng_boilerplate_nestjs
# feat: delete a testimonial

### What does the PR do?
This PR addresses allowing superadmin to delete testimonials by id.

### How should this be manually tested?
To test this feature, you can send a DELETE request to the following endpoint:

`DELETE/api/v1/testimonials/:id`
Authorization: Bearer eyJhbGciOiJIUz

On a successful deletion of the testimonial, the API should return a 200 status code.
Request Example:
[DELETE/api/v1/testimonials/:id]http://localhost:3008/api/v1/testimonials/fe2001be-1bbf-4e50-a81e-5698afbf5e03

## Success Response:
```
{
  "status_code": 200,
  "message": "Testimonial deleted successfully"
}
```

## Error Response:
Unauthorized
```
{
  "message": "User is currently unauthorized, kindly authenticate to continue",
  "status": 401
}
```
Forbidden
```
{
  "message": "You dont have the permission to perform this action",
  "status": 403
}
```

Not Found
```
{
  "message": "Testimonial not found",
  "status": 404
}
```
Tasks
 - [x] Develop server-side logic to handle testimonial deletion requests.
 - [x] Validate incoming testimonial ID data.
 - [x] Validate the user deleting the testimonial is an admin.
Reference Ticket: https://linear.app/team-bingo/issue/BAC2-198/delete-a-testimonial
